### PR TITLE
Add EntityLoadCrossbowEvent

### LINKED ISSUE #4619: CrossbowChargeEvent / BowChargeEvent
**Is your feature request related to a problem? Please describe.**
I'm trying to block players from charging bows in certain regions & prevent them from shooting.
The main issue I have is that the arrows vanish in the players' inventory when a crossbow is charged even though the shooting is canceled. A CrossbowChargeEvent / BowChargeEvent could prevent this (hopefully).

**Additional context**
@JRoy asked me to create this in Discord (id: 194473148161327104)

### PR DESCRIPTION
**Pull request by JRoy**
Tested on the following code:
```java
  @EventHandler
  public void onCharge(EntityChargeCrossbowEvent event) {
    event.setCancelled(true);
    if (event.getEntity() instanceof Player) {
      ((Player) event.getEntity()).updateInventory();
    }
    Bukkit.broadcastMessage("yo");
  }
```

Closes #4619

### CODE REVIEW

**Review by Proximyst — DISMISSED** [review]

**Comment by FazziClay** [comment]
test 1111

File: Spigot-API-Patches/0229-Add-EntityChargeCrossbowEvent.patch [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
++import org.jetbrains.annotations.Nullable;
++
++/**
++ * Called when a LivingEntity charges a crossbow with a projectile.
++ */
++public class EntityChargeCrossbowEvent extends EntityEvent implements Cancellable {
++    private static final HandlerList handlers = new HandlerList();
++    private final ItemStack crossbow;
++    private final EquipmentSlot hand;
++    private boolean canceled;
```
Comment (kadenscott) [inline Spigot-API-Patches/0229-Add-EntityChargeCrossbowEvent.patch:32]: method name is `setCancelled`, variable name is `canceled`

Comment (webcrawls) [inline Spigot-API-Patches/0229-Add-EntityChargeCrossbowEvent.patch]: method name is setCancelled, variable name is canceled

File: Spigot-API-Patches/0229-Add-EntityChargeCrossbowEvent.patch [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
++        return hand;
++    }
++
++    @Override
++    public boolean isCancelled() {
++        return canceled;
++    }
++
++    /**
++     * Set whether or not to cancel the crossbow being charged. If canceled, the
```
Comment (webcrawls) [inline Spigot-API-Patches/0229-Add-EntityChargeCrossbowEvent.patch]: _canceled_

File: Spigot-API-Patches/0229-Add-EntityChargeCrossbowEvent.patch [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
++     *
++     * If set to false, and this event is pertaining to a player entity,
++     * it's recommended that a call to {@link Player#updateInventory()} is made
++     * as the client may think the server still loaded an item into the crossbow.
++     *
++     * @param cancel true if you wish to cancel this event
++     */
++    @Override
++    public void setCancelled(boolean cancel) {
++        this.canceled = cancel;
```
Comment (webcrawls) [inline Spigot-API-Patches/0229-Add-EntityChargeCrossbowEvent.patch]: _canceled_

File: Spigot-API-Patches/0229-Add-EntityChargeCrossbowEvent.patch [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
++     * @return the hand
++     */
++    @NotNull
++    public EquipmentSlot getHand() {
++        return hand;
++    }
++
++    @Override
++    public boolean isCancelled() {
++        return canceled;
```
Comment (webcrawls) [inline Spigot-API-Patches/0229-Add-EntityChargeCrossbowEvent.patch]: _canceled_

**Comment by BillyGalbreath** [comment]
I would suggest renaming this event to `EntityLoadCrossbowEvent` since this is triggered when a crossbow is _loaded_ with a projectile.

Unless I'm missing some magical charge or enchantment that this is for, in which case ignore this comment..

**Review by webcrawls — CHANGES_REQUESTED** [review]
sorry jroy im a stickler for these things 😩

**Comment by JRoy** [comment]
done @BillyGalbreath @kadenscott

**Comment by Prof-Bloodstone** [comment]
Just a FYI comment. We've discussed the naming change on Discord: https://discord.com/channels/289587909051416579/555462289851940864/776374191377088522

The gist is that Charge makes more sense in keeping true to Mojang naming (see NBT for `tag.Charged` and `tag.ChargedProjectiles`) , while using Loaded seems potentially more intuitive. 

We couldn't find a good proper name for it. Both names are good IMO.

**Review by MiniDigger — DISMISSED** [review]

**Review by zachbr — DISMISSED** [review]
The API is called load, the patch and commit are called charge. Please resolve before merging.
I am not overly fussed about which, both have valid reasons to use.

**Comment by JRoy** [comment]
@zachbr fixed api patch to reflect the renamed event

**Comment by MiniDigger** [comment]
@limbo-app ready
(havent touch this yet, just marking this again)

**Review by limbo-app — DISMISSED** [review]
Pull request automatically approved for merging - a core team member has previously approved this pull request.

**Comment by MiniDigger** [comment]
@limbo-app ready

**Review by limbo-app — APPROVED** [review]
Pull request automatically approved for merging - a core team member has previously approved this pull request.

**Comment by MiniDigger** [comment]
thanks!

**Pull request merged**